openapi_python_client/parser/properties.py

```diff
@@ -575,3 +617,27 @@ def property_from_data(
         return _property_from_data(name=name, required=required, data=data)
     except ValidationError:
         return PropertyError(detail="Failed to validate default value", data=data)
+
+
+def type_string(no_optional: bool, prop: Property, inner_type_string: str, is_outer_union: bool = False) -> str:
+    if no_optional or (prop.required and not prop.nullable):
+        if is_outer_union:
+            return f"Union[{inner_type_string}]"
+        return inner_type_string
+    elif prop.nullable and prop.required:
+        if is_outer_union:
+            return f"Optional[Union[{inner_type_string}]]"
+        return f"Optional[{inner_type_string}]"
+    elif not prop.nullable and not prop.required:
```

Comment:
Should this be `elif prop.nullable and not prop.required`, given that we're unioning it with `None` in L632?

Reply:
I dropped this change in the latest commit where I preserve the original typing and just set the default.